examples/vision/3D_image_classification.py

```diff
@@ -0,0 +1,454 @@
+"""
+Title: 3D Image Classification from CT Scans
+Author: [Hasib Zunair](https://hasibzunair.github.io/)
+Date created: 2020/09/10
+Last modified: 2020/09/17
+Description: Train a 3D convolutional neural network to classify location of cancer.
+"""
+"""
+## Introduction
+
+This example will show the steps needed to build a 3D image classification
+model using a 3D convolutional neural network (CNN) to predict the location
+of cancerous regions in CT scans.
+
+## Why 3D CNNs at all?
+
+Traditional use-cases of CNNs are RGB images (3 channels). The goal of
+a 3D CNN is to take as input a 3D volume or a sequence of frames
+(e.g. slices in a CT scan) and extract features from it. A traditional CNN
+extract representations of a single image and puts them in a vector state
+(latent space), whereas a 3D CNN extracts representations of a set of images
+which is required to make predictions from volumetric data (e.g. CT scans).
+A 3D CNN takes the temporal dimension (e.g. 3D context) into account. This way
+of learning representations from a volumetric data is useful to find the right
+label. This is achieved by using 3D convolutions.
+"""
+
+import os
+import zipfile
+import numpy as np
+
+from tensorflow import keras
+from tensorflow.keras import layers
+
+"""
+## Downloading the  NSCLC-Radiomics-Genomics dataset
+
+Since training 3D convolutional neural network are time consuming, a subset of the
+NSCLC-Radiomics-Genomics dataset is used which consists of CT scans
+with gene expression and relevant clinical data. In this example, we will be
+using the "location" attribtute among the available clinical data to build a
+classifier to predict cancerious regions (left or right). Hence, the task is
+a binary classification problem.
+"""
+
+url = "https://github.com/hasibzunair/3D-image-classification-tutorial/releases/download/v0.1/NSCLC-Radiomics-Genomics.zip"
+filename = os.path.join(os.getcwd(), "NSCLC-Radiomics-Genomics.zip")
+keras.utils.get_file(filename, url)
+
+with zipfile.ZipFile("NSCLC-Radiomics-Genomics.zip", "r") as z_fp:
+    z_fp.extractall("./")
+
+"""
+## Load data
+
+The files are provided in Nifti format with the extension .nii. To read the
+scans, the nibabel package is used. To process the data, the following
+is done:
+
+* Volumes are originally rotated by 90 degress, so the orientation is fixed
+* Resize width, height and depth
+
+Here several helper functions are defined to process the data. These functions
+will be used when building training and test datasets.
+"""
+
+import numpy as np
+import nibabel as nib
+import cv2
+
+from scipy.ndimage import zoom
+
+
+def read_nifti_file(filepath):
+    # read file
+    scan = nib.load(filepath)
+    # get raw data
+    scan = scan.get_fdata()
+    # rotate
```

Comment:
Why?

Reply:
The files are in Nifti format which consists of other metadata about the CT scan/study. `nib.load(filepath)` loads the file and then the image data is extracted using `scan.get_fdata()` as a Numpy array. This is what is normally done. Maybe I should add these as comments on the script, no?